src/test/java/co/rsk/bitcoinj/script/StandardRedeemScriptParserTest.java

```diff
@@ -10,156 +10,116 @@
-        }
-
-        Collections.sort(expectedKeysList);
-        Collections.sort(obtainedKeysList);
-
-        assertEquals(expectedKeysList, obtainedKeysList);
+    public void getM_ok() {
+        int expectedM = defaultRedeemScriptKeys.size() / 2 + 1;
+        assertEquals(expectedM, standardRedeemScriptParser.getM());
     }
 
     @Test
-    public void getPubKeys_standard_redeem_script() {
-        Script redeemScript = RedeemScriptUtils.createStandardRedeemScript(btcECKeyList);
-
-        RedeemScriptParser parser = RedeemScriptParserFactory.get(redeemScript.getChunks());
-        List<BtcECKey> obtainedList = parser.getPubKeys();
-
-        List<String> expectedKeysList = new ArrayList<>();
-        for (BtcECKey key : btcECKeyList) {
-            expectedKeysList.add(key.getPublicKeyAsHex());
-        }
-
-        List<String> obtainedKeysList = new ArrayList<>();
-        for (BtcECKey key : obtainedList) {
-            obtainedKeysList.add(key.getPublicKeyAsHex());
-        }
+    public void extractStandardRedeemScriptChunks_ok() {
+        // Arrange
+        Script redeemScript = RedeemScriptUtils.createStandardRedeemScript(defaultRedeemScriptKeys);
+        List<ScriptChunk> expectedRedeemScriptChunks = redeemScript.getChunks();
 
-        Collections.sort(expectedKeysList);
-        Collections.sort(obtainedKeysList);
+        // Act
+        List<ScriptChunk> actualRedeemScripChunks = standardRedeemScriptParser.extractStandardRedeemScriptChunks();
 
-        assertEquals(expectedKeysList, obtainedKeysList);
+        // Assert
+        assertEquals(expectedRedeemScriptChunks, actualRedeemScripChunks);
     }
 
-    @Test(expected = ScriptException.class)
-    public void getPubKeys_invalid_redeem_script() {
-        Script script = ScriptBuilder.createP2SHOutputScript(new byte[20]);
-        RedeemScriptParser parser = RedeemScriptParserFactory.get(script.getChunks());
-
-        parser.getPubKeys();
+    @Test
+    public void getMultiSigType_ok() {
+        Assert.assertEquals(RedeemScriptParser.MultiSigType.STANDARD_MULTISIG, standardRedeemScriptParser.getMultiSigType());
     }
 
-
-
     @Test
-    public void getM_from_multiSig_fast_bridge_redeem_script() {
-        byte[] data = Sha256Hash.of(new byte[]{1}).getBytes();
-        Script fastBridgeRedeemScript = RedeemScriptUtils.createFastBridgeRedeemScript(
-            data,
-            btcECKeyList
-        );
-
-        RedeemScriptParser parser = RedeemScriptParserFactory.get(fastBridgeRedeemScript.getChunks());
+    public void isStandardMultiSig_ok() {
+        Assert.assertTrue(StandardRedeemScriptParser.isStandardMultiSig(standardRedeemScriptParser.extractStandardRedeemScriptChunks()));
+    }
 
-        assertEquals(2, parser.getM());
+    @Test
+    public void isStandardMultiSig_whenEmptyScript_shouldReturnFalse() {
+        Assert.assertFalse(StandardRedeemScriptParser.isStandardMultiSig(new Script(new byte[]{}).getChunks()));
```

Comment:
```suggestion
        Script emptyScript = new Script(new byte[]{});
        Assert.assertFalse(StandardRedeemScriptParser.isStandardMultiSig(emptyScript.getChunks()));
```

nit, but

Reply:
what about using the `RedeemScriptValidator.hasStandardRedeemScriptStructure` directly?
i dont really get the point for the `Parser.is...MultiSig`, is just asking the `RedeemScriptValidator`

Reply:
This is for another task. You right. The plan is to use RedeemScriptValidator directly.